### PR TITLE
[RHPAM-3640] Configure Kafka JBPM Emitter from Operator -7.11.x

### PR DESCRIPTION
**Pull request by desmax74**
Thanks for submitting your Pull Request!
See: https://issues.redhat.com/browse/RHPAM-3640

Please make sure your PR meets the following requirements:

- [x] Pull Request title is properly formatted: `[RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted
- [x] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [x] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by desmax74** [comment]
@spolti

File: kieserver/image.yaml [hunk 1]
```diff
@@ -230,6 +230,24 @@ envs:
...
     description: "Value in milliseconds that indicates how long the 'publish' method will block the operation. Default 2000 milliseconds (2 seconds), this environment variable will be used to set the org.kie.server.jbpm-kafka.ext.max.block.ms system property."
+  - name: "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_ENABLED"
+    example: "true"
+    description: "If true enable the deployment and the configuration of the emitter jar"
+  - name:  "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_ACKS"
+    example: "3"
+    description: "The number of acknowledgments that the emitter requires the leader to have received before considering a request to be complete."
+  - name:  "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_BOOTSTRAP_SERVERS"
+    example: "localhost:9092"
+    description: "A comma separated list of host/port pairs to use for establishing the initial connection to the Kafka cluster, this environment variable will be used to set the org.kie.jbpm.event.emitters.kafka.bootstrap.servers system property"
```
Comment (spolti) [inline kieserver/image.yaml:241]:
```suggestion
    description: "Comma separated list of host/port pairs to use for establishing the initial connection to the Kafka cluster, this environment variable will be used to set the org.kie.jbpm.event.emitters.kafka.bootstrap.servers system property."
```

File: kieserver/image.yaml [hunk 2]
```diff
@@ -230,6 +230,24 @@ envs:
   - name: "KIE_SERVER_KAFKA_EXT_MAX_BLOCK_MS"
     example: "6000"
     description: "Value in milliseconds that indicates how long the 'publish' method will block the operation. Default 2000 milliseconds (2 seconds), this environment variable will be used to set the org.kie.server.jbpm-kafka.ext.max.block.ms system property."
+  - name: "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_ENABLED"
+    example: "true"
+    description: "If true enable the deployment and the configuration of the emitter jar"
```
Comment (spolti) [inline kieserver/image.yaml:235]:
```suggestion
    description: "If true, will configure the JBpm Kafka events integration, when true the kafka bootstrap servers must be set."
```

File: kieserver/image.yaml [hunk 3]
```diff
@@ -230,6 +230,24 @@ envs:
...
+  - name:  "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_ACKS"
+    example: "3"
+    description: "The number of acknowledgments that the emitter requires the leader to have received before considering a request to be complete."
```
[the remark this answers is not in the document]
Comment (spolti) [reply kieserver/image.yaml:238]: also on the other envs you mentioned what system property it does set, maybe it worth to add here as well?

File: kieserver/image.yaml [hunk 4]
```diff
@@ -230,6 +230,24 @@ envs:
...
+  - name:  "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_BOOTSTRAP_SERVERS"
+    example: "localhost:9092"
+    description: "A comma separated list of host/port pairs to use for establishing the initial connection to the Kafka cluster, this environment variable will be used to set the org.kie.jbpm.event.emitters.kafka.bootstrap.servers system property"
+  - name:  "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_CLIENT_ID"
+    example: "client.id"
+    description: "This configuration allows users to set an ID to provide a logical application name for logging purposes, not set by default.This environment variable will be used to set the org.kie.jbpm.event.emitters.kafka.client.id system property"
+  - name:  "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_MAX_BLOCK_MS"
+    example: "2000"
+    description: "Value in milliseconds that indicates how long the 'publish' method will block the operation. Default 2000 milliseconds (2 seconds), this environment variable will be used to set the org.kie.jbpm.event.emitters.kafka.max.block.ms system property."
```
Comment (spolti) [inline kieserver/image.yaml:247]: wouldn't be better replace operation by thread?

Comment (spolti) [reply kieserver/image.yaml:247]:
```suggestion
    description: "Value in milliseconds that indicates how long the 'publish' method will block the operation. Default 2000 milliseconds (2 seconds), this environment variable will be used to set the 'org.kie.jbpm.event.emitters.kafka.max.block.ms' system property."
```

File: kieserver/image.yaml [hunk 5]
```diff
@@ -230,6 +230,24 @@ envs:
...
+  - name:  "KIE_SERVER_KAFKA_JBPM_EVENT_EMITTER_DATE_FORMAT"
+    example: "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+    description: "Date and time format to be sent to Kafka, this environment variable will be used to set the org.kie.jbpm.event.emitters.kafka.date_format system property."
```
Comment (spolti) [inline kieserver/image.yaml:250]: please mention what is the default value.

Comment (spolti) [reply kieserver/image.yaml:250]:
```suggestion
    description: "Date and time format to be sent to Kafka, this environment variable will be used to set the 'org.kie.jbpm.event.emitters.kafka.date_format' system property."
```

**Review by spolti — CHANGES_REQUESTED** [review]

File: kieserver/image.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (desmax74) [reply kieserver/image.yaml:238]: If not set is not passed to kafka, ok to specify sys prop

**Review by spolti — APPROVED** [review]

**Review by spolti — DISMISSED** [review]

**Review by jakubschwan — APPROVED** [review]

**Review by spolti — APPROVED** [review]

**Pull request merged**